File: src/main/java/model/DAO.java
package model;

import model.entity.Schedule;

import java.util.List;

public interface DAO {

    default void saveAll(List<Schedule> scheduleList) {
        scheduleList.forEach(this::save);
    }

    void save(Schedule schedule);

    void update(Schedule schedule);

    void delete(Schedule schedule);

    Schedule readByHash(int hashCode);

    Schedule read(Schedule schedule);
}

File: sl-model/src/main/kotlin/com/example/data/DAO.kt
package com.example.data

import com.example.model.entity.Schedule


interface DAO<Entity, Key> {
    fun create(entity: Entity)
    fun read(key: Key): Entity
    fun update(entity: Entity)
    fun delete(entity: Entity)
    val list: List<Entity>?
}

interface ScheduleReaderDAO {

    fun saveOrUpdate(schedule: Schedule): Boolean
}

